### PR TITLE
[DPE-1917] Updated logs for backup status

### PR DESCRIPTION
**Pull request by dmitry-ratushnyy**
## About

- Added error logs for failed events and info logs for successful events. Events are backup\restore\list backups
- Make a call order of event.defer() and logger.info() consistent

## Related issue

[DPE-1917](https://warthogs.atlassian.net/browse/DPE-1917)

[DPE-1917]: https://warthogs.atlassian.net/browse/DPE-1917?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Mehdi-Bendriss — APPROVED** [review]

File: lib/charms/mongodb/v0/mongodb_backups.py [hunk 1]
```diff
@@ -120,123 +120,141 @@ def _on_s3_credential_changed(self, event: CredentialsChangedEvent):
...
         self._configure_pbm_options(event)
 
     def _on_create_backup_action(self, event) -> None:
         if self.model.get_relation(S3_RELATION) is None:
-            event.fail("Relation with s3-integrator charm missing, cannot create backup.")
-            return
+            return self._fail_event_with_log(
+                event,
+                "Backup failed: relation with s3-integrator charm missing, cannot create backup.",
+            )
```
[the remark this answers is not in the document]
Comment (dmitry-ratushnyy) [reply lib/charms/mongodb/v0/mongodb_backups.py:142]: ok, no problem, can update to calling and then returning

**Review by Mehdi-Bendriss — DISMISSED** [review]

**Review by taurus-forever — DISMISSED** [review]
LGTM.
1) one question inside
2) I cannot comment reordering event.defer() and logger.info. Both should be fine and no reasoning in PR description.

File: lib/charms/mongodb/v0/mongodb_backups.py [hunk 2]
```diff
@@ -182,61 +196,74 @@ def _on_list_backups_action(self, event) -> None:
...
         except (subprocess.CalledProcessError, ExecError) as e:
-            event.fail(f"Failed to list MongoDB backups with error: {str(e)}")
+            self._fail_event_with_log(event, f"List backups failed: {str(e)}")
             return
 
     def _on_restore_action(self, event) -> None:
         if self.model.get_relation(S3_RELATION) is None:
-            event.fail("Relation with s3-integrator charm missing, cannot restore from a backup.")
+            self._fail_event_with_log(
+                event, "Relation with s3-integrator charm missing, cannot restore from a backup."
```
[the remark this answers is not in the document]
Comment (dmitry-ratushnyy) [reply lib/charms/mongodb/v0/mongodb_backups.py:217]: Updated, thank you

**Review by Mehdi-Bendriss — APPROVED** [review]

**Review by taurus-forever — APPROVED** [review]
LGTM, Tnx!

**Pull request merged**